From d4ad14173906db278937a8779dc608a3e155727e Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@gmail.com>
Date: Fri, 25 Mar 2016 13:21:53 +0100
Subject: [PATCH 6/8] Move hrtime_t timestruc_t and timespec_t

hrtime_t timestruc_t and timespec_t should have originally been
included in sys/time.h so lets move them.

longlong_t is not defined by any standard so change it to long long

Signed-off-by: Carlo Landmeter <clandmeter@gmail.com>
Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
Closes #4459
---
 lib/libspl/include/sys/time.h  | 5 +++++
 lib/libspl/include/sys/types.h | 4 ----
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/lib/libspl/include/sys/time.h b/lib/libspl/include/sys/time.h
index f0da440..03da3f7 100644
--- a/lib/libspl/include/sys/time.h
+++ b/lib/libspl/include/sys/time.h
@@ -58,6 +58,11 @@
 #define	NSEC2MSEC(n)    ((n) / (NANOSEC / MILLISEC))
 #endif
 
+typedef	long long		hrtime_t;
+typedef	struct	timespec	timestruc_t;
+typedef	struct	timespec	timespec_t;
+
+
 extern hrtime_t gethrtime(void);
 extern void gethrestime(timestruc_t *);
 
diff --git a/lib/libspl/include/sys/types.h b/lib/libspl/include/sys/types.h
index 25d56be..8996af9 100644
--- a/lib/libspl/include/sys/types.h
+++ b/lib/libspl/include/sys/types.h
@@ -55,10 +55,6 @@ typedef longlong_t	diskaddr_t;
 typedef ulong_t		pgcnt_t;	/* number of pages */
 typedef long		spgcnt_t;	/* signed number of pages */
 
-typedef longlong_t	hrtime_t;
-typedef struct timespec	timestruc_t;
-typedef struct timespec timespec_t;
-
 typedef short		pri_t;
 
 typedef int		zoneid_t;
-- 
2.7.4

